plugins/datadog_tracer/lib/samson_datadog_tracer/samson_plugin.rb

```
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+module SamsonDatadogTracer
+  class Engine < Rails::Engine
+  end
+
+  def self.enabled?
+    !!ENV['DATADOG_TRACER']
+  end
+end
+
+Samson::Hooks.callback :performance_tracer do |klass, method|
+  if SamsonDatadogTracer.enabled?
+    klass.class_eval do
+      include SamsonDatadogTracer::APM
+      helper = SamsonDatadogTracer::APM::Helpers
+      trace_method method
+
+      if method_defined?(method) || private_method_defined?(method)
```

why not just call alias and let it blow up if method is not defined ?

---

also ideally don't class_eval but prepend a module that calls super